### ISSUE TITLE
Make database updates blazing fast

### ISSUE THREAD

**Issue by graue** [graue/potsfyi#31]
tl;dr: they're too slow, make 'em fast. (Original rant follows:)

I just added a commit which prints time taken for create/update using this method:

    start_time = datetime.today()  # at top of create/update function
    # ... function body (including final db commit at the end) ...
    end_time = datetime.today()    # at bottom of create/update function
    print (end_time - start_time).total_seconds()

I tested the time to process 664 tracks on the local SSD on a late-2012 MacBook Air 13". I compared the post-DB overhaul commit 30dff9d, "Print time taken to process a DB update", run with `DEBUG=True ./manage.py update`, and the pre-DB overhaul commit e5dad7a "Fix broken DB_URI option", run with `DEBUG=True ./manage.py createdb`. (To the old commit, I added a timer that works the same as above.) All figures are the average of 5 trials.

* 5.14 sec (avg) to create DB first time, post-overhaul, no existing DB
* 0.62 sec (avg) to update existing DB, post-overhaul, no files changed
* 0.85 sec (avg) to create DB, pre-overhaul

So modest speedup (27%) in the case where nothing changed, but 500% worse for initial create! This is IMHO obviously not a good tradeoff. When nothing's changed we should get a much better speedup than 27%.

I'd like updates with no changes to be at least 50% faster than pre-overhaul creates, and initial creates to be at most 25% slower. Future changes should be guided by speed testing and profiling.

**Comment by graue** [comment]
Eh. I just run updates overnight now. Not the end of the world. Will create a new issue if I have concrete ideas I feel like trying out.
